lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -4297,7 +4297,7 @@ const Js::PropertyRecord* ThreadContext::AddSymbolToRegistrationMap(const char16
     // However, as the key contains a null character we need to hash the symbol name past the null character. The default implementation terminates
     // at the null character, so we use the Js::HashedCharacterBuffer as key. We allocate the key in the recycler memory as it needs to be around
     // for the lifetime of the map.
-    Js::HashedCharacterBuffer<char16> * propertyName = RecyclerNew(GetRecycler(), Js::HashedCharacterBuffer<char16>, stringKey, stringLength);
+    Js::HashedCharacterBuffer<char16> * propertyName = RecyclerNew(GetRecycler(), Js::HashedCharacterBuffer<char16>, propertyRecord->GetBuffer(), propertyRecord->GetLength());
```

Comment:
I looked at this briefly, but can't see why the original string is getting collected. The repro isn't straightforward either. However,  crawler has close to 200 hits so far. I will talk to Curtis about this and see if he has any thoughts.

---
In reply to: [127852232](https://github.com/Microsoft/ChakraCore/pull/3369#discussion_r127852232) [](ancestors = 127852232)